FreeRTOS/FreeRTOS-Plus-TCP
# IPv6/Multi: Reduced complexity FreeRTOS_TCP_WIN.c

Description
-----------
The module FreeRTOS+TCP_WIN.c had 4 functions with a high score on complexity. I cut down their size by moving some code the helper functions:

lTCPWindowRxCheck()  score 66
- prvTCPWindowRx_ExpectedRX()
- prvTCPWindowRx_UnexpectedRX()

lTCPWindowTxAdd()  score 35
- prvTCPWindowTxAdd_FrontSegment()

ulTCPWindowTxGet()  score 43
- pxTCPWindowTx_GetWaitQueue()
- pxTCPWindowTx_GetTXQueue()

prvTCPWindowTxCheckAck()  score 25
- prvTCPWindowTxCheckAck_CalcSRTT()


Test Steps
-----------
I tested by transferring files in both directions while dropping random packets in both directions.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
